Forge.TreeWalker/src/TreeSchemaValidator.cs

```diff
@@ -0,0 +1,275 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="TreeSchemaValidator.cs" company="Microsoft">
+//     Copyright (c) Microsoft Corporation.  All rights reserved.
+// </copyright>
+// <summary>
+//     The TreeSchemaValidator class implements the ITreeSchemaValidator interface.
+// </summary>
+//-----------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Runtime.Serialization.Json;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Forge.DataContracts;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Newtonsoft.Json.Schema;
+
+
+namespace Microsoft.Forge.TreeWalker
+{
+    public class TreeSchemaValidator : ITreeSchemaValidator
+    {
+        private JSchema ForgeSchemaValidationRules;
+        public TreeSchemaValidator() => ForgeSchemaValidationRules = GetRulesFromPath("contracts\\ForgeSchemaValidationRules.json");
```

Comment:
>contracts\\ForgeSchemaValidationRules.json [](http://example.com/codeflow?start=87&length=42)

Users will need to pass in their own rules. When they reference the nuget package in their csproj, they'll also have to include the ForgeSchemaValidationRules.json file in their output if they wish to use it.

We don't have control over where this file lands, so cannot pull it for them automatically. Therefore, we'll need to rely on users passing in the rules, even if they use ForgeSchemaValidationRules.json.